### PR TITLE
Extract sound effects enum into its own file

### PR DESCRIPTION
**Pull request by glebm**
Avoids depending on all of `effects` in places where we only need the enum.

Extracted from #7554

### CODE REVIEW

File: Source/monstdat.h [hunk 1]
```diff
@@ -5,7 +5,9 @@
  */
 #pragma once
 
+#include <cstddef>
 #include <cstdint>
+#include <string>
```
Comment (StephenCWills) [inline Source/monstdat.h:10]: What are these for?

Comment (glebm) [reply Source/monstdat.h:10]: There are a couple of structs at the bottom of this file that use these. They previously worked due to transitive includes from effects.h

Comment (StephenCWills) [reply Source/monstdat.h:10]: I see it now, via `textdat.h`.

**Review by StephenCWills — APPROVED** [review]

**Comment by AJenbo** [comment]
Nice :+1:

**Pull request merged**